File: src/main/scala/org/xbery/artbeams/users/domain/UserRole.scala
package org.xbery.artbeams.users.domain

/**
  * @author Radek Beran
  */
sealed trait UserRole {
  def name: String
}

object UserRole {
  case object RoleUser extends UserRole {
    override val name: String = "ROLE_USER"
  }
}

File: src/main/kotlin/org/xbery/artbeams/users/domain/UserRole.kt
package org.xbery.artbeams.users.domain

/**
 * @author Radek Beran
 */
interface UserRole {
    val name: String

    companion object {
        /**
         * Concrete role for common users.
         */
        object RoleUser : UserRole {
            override val name: String = "ROLE_USER"
        }
    }
}
